cparser/Elab.ml

```diff
@@ -760,15 +760,17 @@ and elab_init_name_group keep_ty loc env (spec, namelist) =
 
 and elab_field_group keep_ty env (Field_group (spec, fieldlist, loc)) =
 
-  let fieldlist = List.map (
-    function
-      | (None, x) -> (Name ("", JUSTBASE, [], loc), x)
-      | (Some n, x) -> (n, x))
-    fieldlist
+  let fieldlist,_ = mmap (fun c id ->
+    match id with
+    | (None, x) ->
+      let name = Printf.sprintf "<anon>__%d" c in
+      (Name (name, JUSTBASE, [], loc), x,true),c+1
+    | (Some n, x) -> (n, x,false),c)
+      0 fieldlist
```

Comment:
This generates unique names within a field group but not within a complete struct or union declaration.

My suggestion would be to use the empty string as name, then in `elab_struct_or_union_info`, onlce the list `m` of members is obtained, scan it again to replace empty names with fresh names.